agent/app/agent_capability.go

```diff
@@ -113,12 +115,17 @@ func (agent *ecsAgent) capabilities() []*ecs.Attribute {
 		}
 	}
 
-	if agent.cfg.TaskCPUMemLimit {
+	// TODO we also need to disable this if agent already managing older tasks
+	if agent.cfg.TaskCPUMemLimit.Enabled() {
 		if _, ok := supportedVersions[dockerclient.Version_1_22]; ok {
 			capabilities = appendNameOnlyAttribute(capabilities, attributePrefix+capabilityTaskCPUMemLimit)
+		} else if agent.cfg.TaskCPUMemLimit == config.ExplicitlyEnabled {
+			// explicitly enabled -- return an error because we cannot fulfil an explicit request
+			return nil, errors.New("engine: Task CPU + Mem limit cannot be enabled due to unsupported Docker version")
 		} else {
-			seelog.Warn("Task CPU + Mem Limit disabled due to unsupported Docker version")
-			agent.cfg.TaskCPUMemLimit = false
+			// implicitly enabled -- don't register the capability, but degrade gracefully
+			seelog.Warn("Task CPU + Mem Limit disabled due to unsupported Docker version. Version 1.22 is required.")
```

Comment:
nit: "or greater"

Or maybe indicate which Docker version is the minimum rather than just the API?

Reply:
I think the API is more appropriate in this context